x-pack/plugins/maps/public/classes/layers/heatmap_layer/heatmap_layer.ts

```diff
@@ -89,81 +81,95 @@ export class HeatmapLayer extends AbstractLayer {
-        prevDataRequest: this.getSourceDataRequest(),
-        requestMeta: buildVectorRequestMeta(
-          this.getSource(),
-          this.getSource().getFieldNames(),
-          syncContext.dataFilters,
-          this.getQuery(),
-          syncContext.isForceRefresh
-        ),
-        syncContext,
-        source: this.getSource(),
-        getUpdateDueToTimeslice: () => {
-          return true;
-        },
-      });
-    } catch (error) {
-      if (!(error instanceof DataRequestAbortError)) {
-        throw error;
-      }
+    const sourceDataRequest = this.getSourceDataRequest();
+    if (!sourceDataRequest) {
+      return false;
     }
+    const sourceData = sourceDataRequest.getData() as MvtSourceData | undefined;
+    if (!sourceData) {
+      return false;
+    }
+
+    return mbSource.tiles?.[0] !== sourceData.urlTemplate;
   }
 
   syncLayerWithMB(mbMap: MbMap) {
-    addGeoJsonMbSource(this._getMbSourceId(), this.getMbLayerIds(), mbMap);
+    this._removeStaleMbSourcesAndLayers(mbMap);
+
+    const sourceDataRequest = this.getSourceDataRequest();
+    const sourceData = sourceDataRequest
+      ? (sourceDataRequest.getData() as MvtSourceData)
+      : undefined;
+    if (!sourceData) {
+      return;
+    }
+
+    const mbSourceId = this.getMbSourceId();
+    const mbSource = mbMap.getSource(mbSourceId);
+    if (!mbSource) {
+      mbMap.addSource(mbSourceId, {
+        type: 'vector',
+        tiles: [sourceData.urlTemplate],
+        minzoom: sourceData.minSourceZoom,
+        maxzoom: sourceData.maxSourceZoom,
+      });
+    }
 
     const heatmapLayerId = this._getHeatmapLayerId();
     if (!mbMap.getLayer(heatmapLayerId)) {
       mbMap.addLayer({
         id: heatmapLayerId,
         type: 'heatmap',
-        source: this.getId(),
+        source: mbSourceId,
+        ['source-layer']: sourceData.layerName,
         paint: {},
       });
     }
 
-    const mbGeoJSONSource = mbMap.getSource(this.getId()) as MbGeoJSONSource;
-    const sourceDataRequest = this.getSourceDataRequest();
-    const featureCollection = sourceDataRequest
-      ? (sourceDataRequest.getData() as FeatureCollection)
-      : null;
-    if (!featureCollection) {
-      mbGeoJSONSource.setData(EMPTY_FEATURE_COLLECTION);
+    const metricFields = this.getSource().getMetricFields();
+    if (!metricFields.length) {
       return;
     }
-
-    const propertyKey = this._getPropKeyOfSelectedMetric();
-    const dataBoundToMap = AbstractLayer.getBoundDataForSource(mbMap, this.getId());
-    if (featureCollection !== dataBoundToMap) {
-      let max = 1; // max will be at least one, since counts or sums will be at least one.
-      for (let i = 0; i < featureCollection.features.length; i++) {
-        max = Math.max(featureCollection.features[i].properties?.[propertyKey], max);
+    const metricField = metricFields[0];
+
+    const tileMetaFeatures = this._getMetaFromTiles();
+    let max = 0;
```

Comment:
I'm wondering if we'll need to be careful for division by 0 issues. If the max ends up being 0 for whatever reason (e.g. sum agg), then the interpolation will not work.

It raises a bigger issue that is is possible that `sum` would be misused here. e.g. summing negative values. If that's the case, we should guard. So I would make ensure the max-value is higher that 0. If not, I would snap to `1`.

The idea that we allow Maps allows the use of `sum` for heatmaps, is because users could have an index-field that stores a count-value (e.g. city population). `sum` would allow them to create a heatmap, not by density of city, but density of population. It's "countable" in that sense.